ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudConfig.java

```diff
@@ -71,6 +83,119 @@
       "com.github.ambry.cloud.azure.ConnectionStringBasedStorageClient";
   public static final String USE_ASYNC_AZURE_APIS = "use.async.azure.apis";
 
+  public static final String AZURE_STORAGE_ACCOUNT_INFO_STR = "storageAccountInfo";
+  public static final String AZURE_STORAGE_ACCOUNT_INFO_NAME = "name";
+  public static final String AZURE_STORAGE_ACCOUNT_INFO_PARTITION_RANGE = "partitionRange";
+  public static final String AZURE_STORAGE_ACCOUNT_INFO_STORAGE_SCOPE = "storageScope";
+  public static final String AZURE_STORAGE_ACCOUNT_INFO_STORAGE_ENDPOINT = "storageEndpoint";
+  public static final String AZURE_STORAGE_ACCOUNT_INFO_STORAGE_CONNECTION_STRING = "storageConnectionString";
+
+  /**
+   * Stores Azure Storage Account related info for one storage account.
+   */
+  public static class StorageAccountInfo {
```

Comment:
nit: Do you think it might be better to have this static class and the parser method to be defined in utility class like `AzureUtils` ? Looks like similar functionality for parsing mysql db info, ZK info, etc. are defined in utility classes like `MySqlUtils`, `ClusterMapUtils` .

Reply:
I initially did this change but after moving the parsing into the constructor of AzureCloudConfig, keeping it in this file made more sense